gridstatus/isone_api/isone_api.py

```diff
@@ -0,0 +1,381 @@
+        """
+        Get the real-time hourly demand data for specified locations and date range.
+
+        Args:
+            date (str): The start date for the data request. Use "latest" for most recent data.
+            end_date (str | None): The end date for the data request. Only used if date is not "latest".
+            locations (list[str], optional): List of specific location names to request data for.
+                                             If None, data for all locations will be retrieved.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the real-time hourly demand data for all requested locations.
+        """
+        all_data = []
+
+        match (date, locations):
+            case ("latest", None):
+                url = f"{BASE_URL}/realtimehourlydemand/current"
+                response = self.make_api_call(url)
+                df = pd.DataFrame(response["HourlyRtDemands"]["HourlyRtDemand"])
+                df["LocId"] = df["Location"].apply(lambda x: x["@LocId"])
+                df["Location"] = df["Location"].apply(lambda x: x["$"])
+                return self._handle_demand(df, interval_minutes=60)
+
+            case ("latest", _):
+                for location in locations:
+                    location_id = ZONE_LOCATIONID_MAP.get(location)
+                    if not location_id:
+                        raise ValueError(
+                            f"{location}: Not a known ISO NE Hub or Zone for this data",
+                        )
+                    url = f"{BASE_URL}/realtimehourlydemand/current/location/{location_id}"
+                    response = self.make_api_call(url)
+                    data = response["HourlyRtDemand"]
+                    if not data:
+                        raise NoDataFoundException(
+                            f"No data found for location: {location}",
+                        )
+                    data["Location"] = location
+                    data["LocId"] = location_id
+                    all_data.append(data)
+
+            case _:
+                if not locations:
+                    locations = [
+                        loc
+                        for loc in ZONE_LOCATIONID_MAP.keys()
+                        if loc not in EXCLUDE_FROM_REALTIME_HOURLY_DEMAND
+                    ]
+
+                for location in locations:
+                    location_id = ZONE_LOCATIONID_MAP.get(location)
+                    if (
+                        not location_id
+                        or location in EXCLUDE_FROM_REALTIME_HOURLY_DEMAND
+                    ):
+                        raise ValueError(
+                            f"{location}: Not a known ISO NE Hub or Zone for this data",
+                        )
+
+                    url = f"{BASE_URL}/realtimehourlydemand/day/{date.strftime('%Y%m%d')}/location/{location_id}"
+                    response = self.make_api_call(url)
+                    print(response)
```

Comment:
Should we switch this to a log? Or maybe remove if this was for debugging.

Reply:
yeah debugging. Maybe I will re-request review once ready, I was going to add a few more test cases to cover more of the code paths in the methods.